maud_macros/src/lib.rs

```diff
@@ -54,3 +68,22 @@ fn expand(input: TokenStream) -> TokenStream {
         maud::PreEscaped($output_ident)
     })
 }
+
+#[cfg(feature = "streaming")]
+fn expand_stream(input: TokenStream) -> TokenStream {
+    let output_ident = TokenTree::Ident(Ident::new("__maud_output", Span::def_site()));
+    let markups = match parse::parse(input) {
+        Ok(markups) => markups,
+        Err(()) => Vec::new(),
+    };
+    let stmts = generate::generate_stream(markups, output_ident.clone());
+    quote!({
+        extern crate maud;
+        extern crate futures;
+        let mut $output_ident: futures::stream::FuturesOrdered<
+            Box<futures::Future<Item = &str, Error = &str> + Send>
```

Comment:
I think this should be `Box<futures::Future<Item = maud::Render, Error = maud::Render>>`, such that you can use anything that implements `maud::Render`. However I seem to be hitting the limits of my Rust skills as I can't see a way to make it compile. @lfairy do you have any ideas about how we might go about this?

Reply:
I believe that would have to be `Markup` or `Box<dyn maud::Render>` instead, as otherwise the consumer wouldn't know how much memory to allocate for each item in the stream?